engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1359,7 +1409,8 @@ private void addDAGToCurrentDAG(ModuleMeta moduleMeta)
     String name;
     for (OperatorMeta operatorMeta : subDag.getAllOperators()) {
       name = subDAGName + MODULE_NAMESPACE_SEPARATOR + operatorMeta.getName();
-      this.addOperator(name, operatorMeta.getOperator());
+      Operator op = this.addOperator(name, operatorMeta.getOperator());
```

Comment:
@tweise should we drop "final" qualifier from name and id to avoid the complexity of copying from the original OperatorMeta? If not, I suggest introducing copy constructor for OperatorMeta that takes original OperatorMeta as a parameter and encapsulates the copy operation.